src/keyboard.rs

```diff
@@ -71,55 +71,179 @@
 
 use smol_str::SmolStr;
 
+/// Represents the current state of the keyboard modifiers
+///
+/// Each flag represents a modifier and is set if this modifier is active.
+#[derive(Default, Debug, Clone, Copy, PartialEq, Eq, Hash)]
+pub struct ModifiersState {
+    pub(crate) flags: ModifiersStateFlags,
```

Comment:
I'm not sure how `PartialEq` and `Eq` should work here. Internally in winit we want to compare by each field, but users might not want such a thing.